.github/workflows/sm-cipher.yml

```diff
@@ -0,0 +1,257 @@
+          echo "WOLFSM_ROOT=$(realpath  "$GITHUB_WORKSPACE/../wolfsm")"   >> "$GITHUB_ENV"
+          echo "WOLFSSL_ROOT=$(realpath "$GITHUB_WORKSPACE/../wolfssl")"  >> "$GITHUB_ENV"
+
+          echo "GITHUB_ENV=$GITHUB_ENV"
+
+          echo "contents..."
+          # typically "/home/runner/work/wolfssl/wolfssl" contains wolfssl source
+          pwd
+          ls
+
+      - name: Get wolfssl
+        run: |
+          # We are in wolfsm repo, fetch wolfssl code
+
+          # Show our custom values:
+          echo "GITHUB_WORK            = $GITHUB_WORK"
+
+          # WOLFSM_ROOT is the repo root for wolfsm clone
+          echo "WOLFSM_ROOT            = $WOLFSM_ROOT"
+
+          echo "Start pwd:"
+          pwd
+          # we're typically in $GITHUB_WORKSPACE=/home/runner/work/wolfssl/wolfssl
+          # goto /home/runner/work to fetch wolfsm
+
+          echo "Current pwd for wolfsm clone fetch: $(pwd)"
+          GITHUB_WORK=$(realpath "$GITHUB_WORKSPACE/../..")
+          echo "GITHUB_WORKSPACE=$GITHUB_WORKSPACE"
+
+
+          pushd ../
+            echo "Updated pwd for wolfssl clone fetch: $(pwd)"
+
+            echo "clone --depth 1 https://github.com/$REPO_OWNER/wolfssl.git wolfssl"
+
+            git   clone --depth 1 https://github.com/$REPO_OWNER/wolfssl.git wolfssl
+
+            cd ./wolfssl
+            echo "Contents of this path for wolfssl = $(pwd)"
+            ls
+          popd
+
+          # ** END ** Get wolfssl
+
+      - name: Install wolfsm
+        run: |
+          # Run the local install.sh install script to install wolfsm code
+
+          echo "Current pwd for wolfsm clone fetch: $(pwd)"
+          GITHUB_WORK=$(realpath "$GITHUB_WORKSPACE/../..")
+          echo "GITHUB_WORKSPACE=$GITHUB_WORKSPACE"
+
+          # Typically /home/runner/work
+          echo "GITHUB_WORK=$GITHUB_WORK"
+          pwd
+          echo "pushd $WOLFSM_ROOT"
+          pushd "$WOLFSM_ROOT"
+            pwd
+            ls
+
+            echo "wolfssl check"
+            ls ../wolfssl
+
+            echo "Call wolfsm/install.sh to install wolfsm code into $WOLFSSL_ROOT"
+            ./install.sh "$WOLFSSL_ROOT"
+          popd
+
+          echo "contents..."
+          pwd
+          ls
+
+          # Done with install wolfssl
+
+      - name: Compile wolfssl
+        run: |
+          # Compile fresh wolfSSL with wolfsm code
+
+          cd "$WOLFSSL_ROOT"
+          echo "Current directory:           $PWD"
+
+          ./autogen.sh
+          ./configure --enable-sm3 --enable-sm4-ecb --enable-sm4-cbc --enable-sm4-ctr --enable-sm4-gcm --enable-sm4-ccm --enable-sm2
+          make
+          # Done with compile wolfssl
+
+      - name: Test SM wolfcrypt
+        shell: bash
+        run: |
+          # Run client / server tests from cloned wolfssl directory
+
+          cd "$WOLFSSL_ROOT"
+          echo "Current directory:           $PWD"
+
+          set -euo pipefail
+
+          ./wolfcrypt/test/testwolfcrypt
```

Comment:
Should be able to do a 'make check'.

Reply:
Good idea, added.

Reply:
Both were only for example clarity and comprehensive testing.  I agree they were duplicates. 

I've pushed a new commit removing the other testwolfcrypt and TLS tests.

See [example](https://github.com/gojimmypi/wolfsm/actions/runs/17844203639/job/50740254310) on my branch.